src/test/java/com/thebois/models/world/WorldTests.java

```diff
@@ -177,4 +165,96 @@ private boolean matrixEquals(final ITile[][] worldMatrix1, final ITile[][] world
         return true;
     }
 
+    private static Stream<Arguments> getCorrectCoordinatesToTest() {
+        return Stream.of(Arguments.of(0, 0, 2, 2),
+                         Arguments.of(5, 5, 10, 10),
+                         Arguments.of(0, 0, 0, 0),
+                         Arguments.of(20, 20, 11, 11),
+                         Arguments.of(49, 49, 40, 40)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("getCorrectCoordinatesToTest")
+    public void testFindNearestStructure(final int startX,
+                                         final int startY,
+                                         final int endX,
+                                         final int endY) {
+
+        // Arrange
+        final World world = new World(50, 10);
+        world.createStructure(endX, endY);
+
+        // Act
+        final Optional<IStructure> foundStructure = world.findNearestStructure(new Position(
+            startX,
+            startY));
+
+        // Assert
+        if (foundStructure.isPresent()) {
+            assertThat(foundStructure.get().getPosition()).isEqualTo(new Position(endX, endY));
+        }
+        else {
+            assertThat(foundStructure.isEmpty()).isEqualTo(false);
+        }
```

Comment:
You shouldn't have conditional assertions, because you arranged the test so you already know what the results should be.

`Optional` has a method called `orElseThrow` that gets the value, and if it isn't present it throws an exception (thus failing the test). 

If you are in fact using a dataset that contains two different things to test, when there is a found structure and when there isn't one, then you've got two tests there. Not one.